main.js

```diff
@@ -0,0 +1,310 @@
+  if (locationName == defaultLocationName) {
+    $('.not-default').addClass('hide');
+    $('.is-default').removeClass('hide');
+  } else {
+    $('.not-default').removeClass('hide');
+    $('.is-default').addClass('hide');
+  }
+  //remove the ', USA' from cities in the United States to improve display
+  if (locationName.search(/,\sUSA/) != -1) {
+    locationName = locationName.slice(0, locationName.search(/,\sUSA/));
+  }
+  locationLatitude = data.results[0].geometry.location.lat;
+  locationLongitude = data.results[0].geometry.location.lng;
+  getDailyWeather(locationName, locationLatitude, locationLongitude);
+  getWeeklyWeather(locationLatitude, locationLongitude);
+}
+
+var renderDailyForecast = function() {
+  var source = $('#today-forecast-template').html()
+  var template = Handlebars.compile(source)
+  var newHtml = template(today)
+  $('.daily-forecast').html(newHtml)
+}
+
+var renderWeeklyForecast = function() {
+  $('.weekly-forecast').html("")
+  var source = $('#five-day-forecast-template').html()
+  var template = Handlebars.compile(source)
+  for (i = 0; i < 5; i++) {
+    newHtml = template(weather[i])
+    $('.weekly-forecast').append(newHtml)
+  }
+}
+
+//generate the today object with current weather data
+var currentWeather = function(data, location) {
+  today = {};
+
+  var temp = function() {
+    if (data.main.temp) {
+      return Math.round(data.main.temp);
+    } else {
+      return null;
+    }
+  };
+
+  var cityName = function() {
+    if (location) {
+      return location;
+    } else {
+      return null;
+    }
+  };
+
+  var weatherDescription = function() {
+    if (data.weather[0].description) {
+      return data.weather[0].description;
+    } else {
+      return null;
+    }
+  };
+
+  var icon = function() {
+    if (data.weather[0].icon) {
+      var icon = data.weather[0].icon;
+      return icon
+    } else {
+      return null;
+    }
+  };
+
+  today = {
+    temp: temp(),
+    cityName: cityName(),
+    weatherDescription: weatherDescription(),
+    icon: icon()
+  };
+  renderDailyForecast();
+};
+
+//generate the weather array with the 5-day-forecast weather data
+var weeklyWeather = function(data) {
+  //determine current day
+  var now = moment()
+  if (now.hour() > 12) {
+    previousDay = now.day()
+  } else var previousDay = now.day() - 1
+  weather = [];
+  //for each data point in the forecast list from the API, check whether that date is after the current day AND greater than or equal to 12pm, increment previous day each loop
+  for (j = 0; j < data.list.length; j++) {
+    //date and time data from the API, converted into a moment object
+    dt = moment(moment.unix(data.list[j].dt))
+    //numeric day of the API data item
+    var dataDay = dt.day()
+    //timestamp of the API item
+    var dataTime = dt.hour()
+    if (dataDay - previousDay == 1 || dataDay - previousDay == -6) {
```

Comment:
The last 3 comments aren't needed as much as a comment here... if you came back to this code, would you know why it was 1 or -6?